src/accelerate/data_loader.py

```diff
@@ -692,6 +696,7 @@ def prepare_data_loader(
             new_dataset,
             device=device if put_on_device and state.distributed_type != DistributedType.TPU else None,
             sampler=new_batch_sampler,
+            batch_size=getattr(dataloader, "batch_size", _PYTORCH_DATALOADER_KWARGS["batch_size"]),
```

Comment:
If `(batch_size != 1 and batch_size is not None) and sampler_is_batch_sampler`, is there any "default" way to shard batches without explicit instruction from user? I think at least a warning should be raised.

Reply:
BTW, `hasattr(dataloader, "batch_size")` is always `True` for `DataLoader`.

Reply:
> BTW, `hasattr(dataloader, "batch_size")` is always `True` for `DataLoader`.

resolving this in the other PR. wrt warning, current logic follows PyTorch API and as such warning isn't warranted.